### PR TITLE
[iOS] Add `isBackgroundPlaybackEnabled` to `PlaybackConfig`

### PR DESCRIPTION
**Pull request by stonko1994**
### Changes
This PR adds the possibility of configuring background playback. For now, this is only supported on iOS. Android will follow at a later point. 

To enable background playback, set `PlaybackConfig.isBackgroundPlaybackEnabled` to `true` and configure the App accordingly.

### CODE REVIEW

File: example/ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -8,6 +9,13 @@ internal class AppDelegate: FlutterAppDelegate {
         _ application: UIApplication,
         didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
     ) -> Bool {
+        // configure for background audio
+        do {
+            try AVAudioSession.sharedInstance().setCategory(.playback)
+        } catch {
+            print("Setting category to AVAudioSessionCategoryPlayback failed.")
+        }
+
```
Comment (stonko1994) [inline example/ios/Runner/AppDelegate.swift:18]: This is a very similar version of configuring background playback on iOS. There will be a separate PR where this setup will be moved to the flutter part of the example.

File: example/lib/pages/basic_playback.dart [hunk 2]
```diff
@@ -25,6 +25,7 @@ class _BasicPlaybackState extends State<BasicPlayback> {
       key: Env.bitmovinPlayerLicenseKey,
       playbackConfig: PlaybackConfig(
         isAutoplayEnabled: true,
+        isBackgroundPlaybackEnabled: true,
```
Comment (hawk23) [inline example/lib/pages/basic_playback.dart:28]: Sidenote: When you add support to configure background playback from dart side, it would be good to have a separate sample for it to not bloat basic playback

**Review by hawk23 — APPROVED** [review]
Flawless 🎉

File: example/lib/pages/basic_playback.dart (hunk 2 above)
Comment (stonko1994) [reply example/lib/pages/basic_playback.dart:28]: Agree yes 👍 Will keep that in mind.
